src/geometry/algorithms/ArborX_Intersects.hpp

```diff
@@ -412,6 +412,70 @@ struct intersects<PointTag, TetrahedronTag, Point, Tetrahedron>
   }
 };
 
+template <typename Segment>
+struct intersects<SegmentTag, SegmentTag, Segment, Segment>
+{
```

Comment:
Also you definitely should ensure this cannot be instantiated for a dimension other than 2.

Reply:
Imho, it's fine to instantiate it for other dimensions and error out due to
```
    static_assert(GeometryTraits::dimension_v<Segment> == 2);
```

Reply:
I see it now